src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,54 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static int AGE_AT_LEAST_REQUIRE = 18;
+    private static int LENGTH_AT_LEAST_REQUIRE = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkNull(user);
+        checkAge(user.getAge());
+        checkLogin(user.getLogin());
+        checkPassword(user.getPassword());
+        storageDao.add(user);
+        return user;
+    }
+
+    private void checkAge(Integer age) {
+        if (age < AGE_AT_LEAST_REQUIRE) {
+            throw new RegistrationException("Not allow register too young user");
+        }
+    }
+
+    private void checkPassword(String password) {
+        if (password == null) {
+            throw new RegistrationException("Password can't be null");
+        }
+        if (password.length() < LENGTH_AT_LEAST_REQUIRE) {
+            throw new RegistrationException("Password can't be to short");
```

Comment:
The exception message 'Password can't be to short' contains a typo and is not informative. It should be 'Password can't be too short. Minimum length required is ' + LENGTH_AT_LEAST_REQUIRE.